File: t_news.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*

'''
早报
早报地址：https://www.163.com/dy/media/T1603594732083.html
cron 20 8 * * * t_news.py
'''

import requests
from lxml import etree

def main():
    url="https://www.163.com/dy/media/T1603594732083.html"
    rsp=requests.get(url)
    html=etree.HTML(rsp.text)
    today_url=html.xpath("//h2[@class='media_article_title']/a/@href")[0]
    rsp=requests.get(today_url)
    html=etree.HTML(rsp.text)
    news_list=html.xpath("//div[@class='post_body']/p[2]//text()")
    news_list=news_list[1:]

    news="\n".join(str(i) for i in news_list)
    print(news)
    try:
        from sendNotify import send
        send("早报", news)
    except:
        print("加载通知服务失败~")


if __name__ == "__main__":
    main()

File: t_news.js
/*
cron 20 8 * * * t_news.js
new Env('早报');
*/

const request = require('request');
const notify = require('./sendNotify');

const name = '早报'
const url = 'https://www.163.com/dy/media/T1603594732083.html'

!(async () => {
    try {
        request(url, async (err, res) => {
            // let list = res.body.match(/<a class="img" href="(.*)">/g)
            // list[0].match(/<a class="img" href="(.*)">/g)
            // news_url = RegExp.$1
            const news_url = res.body.match(/https:\/\/www.163.com\/dy\/article.*html/g)[1];

            request(news_url, async (err, res) => {
                res.body.match(/<div class="post_body">[^<]*?<p class="f_center">.*?<p id=".*?">.*?<br\/>(.*?)<\/p>/g)[0]
                let news = RegExp.$1.replace(/<br\/>/g, '\n')
                console.log(news)
                await notify.sendNotify(`${name}`, news)
            })
        })
    } catch (error) {
        console.log(`error: ${error}`)
        await notify.sendNotify(`${name}`, '数据抓取失败!')
    }
})();

